src/hotspot/share/utilities/utf8.cpp

```diff
@@ -392,6 +392,63 @@ bool UTF8::is_legal_utf8(const unsigned char* buffer, int length,
   return true;
 }
 
+// Return true if `b` could be the starting byte of an encoded 2,3 or 6
+// byte sequence.
+static bool is_starting_byte(unsigned char b) {
+  return b >= 0xC0 && b <= 0xEF;;
```

Comment:
Do you plan to use this method only for modified UTF-8 or for standard Utf-8 as well? Standard UTF-8 also uses F0-F7 as starting bytes.

Also, remove the double semicolon.

Reply:
AFAIK the VM only deals with modified UTF-8.

Reply:
;; fixed